pkgs/applications/science/logic/stp/default.nix

```
@@ -36,30 +41,62 @@ stdenv.mkDerivation rec {
     })
   ];
 
-  postPatch = ''
-    # Upstream fix for gcc-13 support:
-    #   https://github.com/stp/stp/pull/462
-    # Can't apply it as is as patch context changed in ither patches.
-    # TODO: remove me on 2.4 release
-    sed -e '1i #include <cstdint>' -i include/stp/AST/ASTNode.h
-  '';
-
   buildInputs = [
     boost
     zlib
     minisat
     cryptominisat
     python3
+    gmp
   ];
-  nativeBuildInputs = [ cmake bison flex perl ];
+  nativeBuildInputs = [ cmake bison flex perl pkg-config gtest lit ];
+  cmakeFlags = [ "-DBUILD_SHARED_LIBS=ON" "-DENABLE_TESTING=ON" "-DLIT_ARGS=-v" ];
   preConfigure = ''
     python_install_dir=$out/${python3.sitePackages}
     mkdir -p $python_install_dir
     cmakeFlagsArray=(
       $cmakeFlagsArray
-      "-DBUILD_SHARED_LIBS=ON"
       "-DPYTHON_LIB_INSTALL_DIR=$python_install_dir"
     )
+
+    # Version is wrong in 2.3.4, fix it.
+    substituteInPlace CMakeLists.txt \
+      --replace-fail 2.3.3 ${version} \
+      --replace-fail GIT-hash-notfound ${version}
+
+    # We want to use the Nix wrapper for the output check tool instead of running it through Python
+    substituteInPlace tests/query-files/lit.cfg \
+      --replace-fail "pythonExec + ' ' +OutputCheckTool" "OutputCheckTool"
+
+    # Link in gtest and the output check utility.
+    mkdir -p deps
+    ln -s ${gtest.src} deps/gtest
+    ln -s ${outputcheck} deps/OutputCheck
+  '';
+
+  doCheck = true;
+
+  postInstall = ''
+    # Clean up installed gtest/gmock files that shouldn't be there.
+    find $out/ \( -iname '*gtest*' -o -iname '*gmock*' \) -print | while read -r file; do
```

could a globstar be just as effective?

```suggestion
    shopt -s globstar
    for file in $out/**/*gtest* $out/**/*gmock*; do
```

---

Likely so!

---

perhaps `shopt -s nocaseglob`?

---

Learning all the shopt today. Thank you!